dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/LambdaEventDAO.java

```diff
@@ -39,6 +51,44 @@ public void delete(LambdaEvent lambdaEvent) {
         session.flush();
     }
 
+    private List<Predicate> processQuery(String filter, String sortCol, String sortOrder, CriteriaBuilder cb, CriteriaQuery query, Root<LambdaEvent> event) {
+        List<Predicate> predicates = new ArrayList<>();
+        if (!Strings.isNullOrEmpty(filter)) {
+            predicates.add(
+                    // ensure we deal with null values and then do like queries on those non-null values
+                    cb.or(cb.and(cb.isNotNull(event.get("message")), cb.like(cb.upper(event.get("message")), "%" + filter.toUpperCase() + "%")), //
+                            cb.and(cb.isNotNull(event.get("githubUsername")), cb.like(cb.upper(event.get("githubUsername")), "%" + filter.toUpperCase() + "%")), //
```

Comment:
Should these be "and" or "or"? I don't know CB very well, so I may be reading this work, but it seems like if you pass `coverbeck` as a filter, it would require githubUsername, reference, type, etc. to all be `coverbeck`. But I may be reading this wrong.

Regardless of whether I'm right or wrong, it would be good to have a test above that test an actual filter value -- at a skim, I think all the tests are always passing `null` as the arguments.

Reply:
Ah, I see now, `cb.or` takes multiple arguments -- I misread the parens.

Reply:
Added test with a filter value